ports/servoshell/desktop/headed_window.rs

```diff
@@ -446,6 +444,26 @@ impl Window {
 }
 
 impl WindowPortsMethods for Window {
+    fn screen_geometry(&self) -> ScreenGeometry {
+        let hidpi_factor = self.hidpi_factor();
+        let toolbar_size = Size2D::new(0.0, (self.toolbar_height.get() * self.hidpi_factor()).0);
+
+        let screen_size = self.screen_size.to_f32() * hidpi_factor;
+        let availabe_screen_size = screen_size - toolbar_size;
+
+        // Offset the WebView origin by the toolbar so that it reflects the actual viewport and
+        // not the window origin.
+        let window_origin = self.winit_window.outer_position().unwrap_or_default();
```

Comment:
Should we use `inner_position()` here? Since we are adding the toolbar height to calculate `offset`, I assume we would also want the window decorations like the window border to be taken into account as well. But it looks like the earlier code was already using `outer_position` so not sure if this is a problem.

Reply:
I think you're right. I've modified this to user `inner_position()`.